new-log-viewer/src/services/LogExportManager.ts

```diff
@@ -0,0 +1,70 @@
+import {downloadBlob} from "../utils/file";
+
+
+const EXPORT_LOG_PROGRESS_INITIALIZATION = 0;
+const EXPORT_LOG_PROGRESS_COMPLETE = 1;
+
+/**
+ * Manager for exporting logs to a file.
+ */
+class LogExportManager {
+    /**
+     * Internal buffer which stores decoded chunks of log data.
+     */
+    readonly #chunks: string[] = [];
+
+    /**
+     * Total number of chunks to export.
+     */
+    readonly #numChunks: number;
+
+    /**
+     * Name of the file to export to.
+     */
+    readonly #exportedFileName: string;
+
+    constructor (numChunks: number, fileName: string) {
+        this.#numChunks = numChunks;
+        this.#exportedFileName = `exported-${fileName}-${new Date().toISOString()
+            .replace(/[:.]/g, "-")}.log`;
+    }
+
+    /**
+     * Append the provided chunk of logs into an internal buffer.
+     * If the number of chunks reaches the specified limit, trigger a download.
```

Comment:
* Function docstrings should be written as if they're prefixed by "This function ..." so in this case the docstring should change to "Append**s** ..."
* Wrap to 100 (follow Markdown formatting).